tools/distpartitioning/utils.py

```diff
@@ -701,11 +705,47 @@ def generate_read_list(num_files, world_size):
     Returns:
     --------
     read_list : np.array
-        Array of target file IDs to read.
+        Array of target file IDs to read. Each worker is expected
+        to read the list of file indexes in its rank's index in the list.
+        e.g. rank 0 reads the file indexed in read_list[0], rank 1 the
+        ones in read_list[1] etc.
+
 
     Examples
     --------
     >>> tools.distpartitionning.utils.generate_read_list(10, 4)
     [array([0, 1, 2]), array([3, 4, 5]), array([6, 7]), array([8, 9])]
     """
     return np.array_split(np.arange(num_files), world_size)
+
+
+def generate_roundrobin_read_list(num_files, world_size):
```

Comment:
how about combining these 2 different generation method into 1 function with an argument like `strategy=`?

Reply:
Can we have a decision here? I think I also prefer the two methods.

Reply:
sure, let's keep 2 methods.